Keyword `not` (#147)
Intelligence-AI/learnjsonschema.com: content/2020-12/applicator/not.markdown

  - vocabulary: applicator
    keyword: else
---

The `not` keyword is used to declare that an instance only validates if it doesn't validate against the given subschema. It is essentially a way to define a rule that an instance should not match.

* The value of this keyword must be a valid JSON Schema.

> * The boolean `false` schemas, are typically an alias to `{ not: {} }`.

Reviewer: ```suggestion
* The boolean `false` schema, can be thought as an alias to `{ not: {} }`.
```

Annotations are dropped when an instance fails. Therefore, in the case of `not`, annotations are always dropped because:

1. If the subschema of `not` passes (producing annotations), then not itself fails, resulting in the annotations being dropped.
2. If the subschema of `not` fails, no annotations are produced, and there is nothing for not to pass on.

## Examples

{{<schema `Schema with 'not' keyword`>}}
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "not": {
    "type": "string"
  }
}
{{</schema>}}

{{<instance-pass `An instance with values other than string is valid`>}}
77
{{</instance-pass>}}

{{<instance-fail `An instance with a string value is invalid`>}}
"foo"
{{</instance-fail>}}

{{<schema `Schema with 'not' set to false`>}}
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",

>   "not": false

Reviewer: Fun example :)
